capstone-backend/src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -125,41 +153,67 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     JsonArray itemsInfo = items.getAsJsonArray();
 
     for (JsonElement book : itemsInfo) {
-      JsonObject bookInfo = book.getAsJsonObject();
-      String id = bookInfo.get("id").getAsString();
-
-      // volumeInfo houses all of the information about the book itself: description,
-      // authors, title
-      JsonElement volumeInfo = bookInfo.get("volumeInfo");
-      if (volumeInfo == null)
-        continue;
-      JsonObject volumeInfoObj = volumeInfo.getAsJsonObject();
-
-      JsonElement titleElement = volumeInfoObj.get("title");
-      String title = titleElement != null ? titleElement.getAsString() : "";
-
-      // there may be any number of authors, not guaranteed to only have 1
-      JsonElement authors = volumeInfoObj.get("authors");
-      ArrayList<String> authorNames = new ArrayList<>();
-      if (authors != null) {
-        JsonArray authorsArray = authors.getAsJsonArray();
-        for (JsonElement name : authorsArray) {
-          authorNames.add(name.getAsString());
-        }
+      VolumeData bookObject = individualBookToVolumeData(book);
+      if (bookObject != null) {
+        volumes.add(individualBookToVolumeData(book));
+      }
+    }
+    return volumes;
+  }
+
+  /*
+   * Takes in an individual book response from the Google Books API and converts
+   * it to a VolumeData object.
+   * 
+   * @param singleBook a single book from the Books API as a JsonElement
+   * 
+   * @return a singular object of the book from the API call
+   */
+  static VolumeData individualBookToVolumeData(JsonElement singleBook) {
+    JsonObject bookInfo = singleBook.getAsJsonObject();
+    String id = bookInfo.get("id").getAsString();
+
+    // volumeInfo houses all of the information about the book itself: description,
+    // authors, title
+    JsonElement volumeInfo = bookInfo.get("volumeInfo");
+    if (volumeInfo == null)
+      return null;
+    JsonObject volumeInfoObj = volumeInfo.getAsJsonObject();
+
+    JsonElement titleElement = volumeInfoObj.get("title");
+    String title = titleElement != null ? titleElement.getAsString() : "";
```

Comment:
Should this return null in the case that there is no title?

Reply:
That's a good idea! If it doesn't have a non empty title, I don't think it should be returned since all books should at the very least have a title